src/Converter.cc

```diff
@@ -657,16 +662,24 @@ void Converter::Remove(tinyxml2::XMLElement *_elem,
 
   if (attributeName)
   {
-    _elem->DeleteAttribute(attributeName);
+    const char * attributeValue = _elem->Attribute(attributeName);
+    if (!_removeEmpty || (attributeValue && attributeValue[0] == '\0'))
```

Comment:
nit: This works fine, but I think using `std::strlen(attributeValue) == 0 ` would be more easily readable

Reply:
https://github.com/ignitionrobotics/sdformat/pull/720/commits/be159db6bfc592edecc05cfd64bcc16701716eb2